src/one/nio/net/native/socket.c

```diff
@@ -520,34 +535,69 @@ Java_one_nio_net_NativeSocket_setTimeout(JNIEnv* env, jobject self, jint timeout
     setsockopt(fd, SOL_SOCKET, SO_SNDTIMEO, &tv, sizeof(tv));
 }
 
+JNIEXPORT jint JNICALL
+Java_one_nio_net_NativeSocket_getTimeout(JNIEnv* env, jobject self) {
+    int fd = (*env)->GetIntField(env, self, f_fd);
+    struct timeval tv;
+    socklen_t len = sizeof(tv);
+    if(getsockopt(fd, SOL_SOCKET, SO_RCVTIMEO, &tv, &len)== 0){
+       return (tv.tv_sec * 1000) + (tv.tv_usec / 1000);
```

Comment:
Isn't there a compiler warning about implicit casting `long` to `jint`?

Reply:
I think the warning needs to be explicitly enabled with -Wconversion other-wize no warning.

The value can be set via the setTimeout(int timeout) imho it makes sense for getTimeout to return int and not long.